### PR TITLE
Avro 3584 rust unit test

### PR DESCRIPTION
**Pull request by clesaec**
[AVRO-3584](https://issues.apache.org/jira/browse/AVRO-3584) : Add unit test for Rust.

### CODE REVIEW

**Comment by martin-g** [comment]
You may need to rebase your Git clone. There are 10 merge commits which should not be there.

File: lang/rust/avro/src/schema.rs [hunk 1]
```diff
@@ -2030,6 +2030,51 @@ mod tests {
...
+            .clone();
+        let schema_b = list
+            .get(1)
+            .unwrap()
+            .clone();
+
+        match schema_a {
+            Schema::Record { fields, .. } => {
+                let f1 = fields.get(0);
+                let string = f1.unwrap().schema.canonical_form();
```
Comment (martin-g) [inline lang/rust/avro/src/schema.rs:2066]: `string` looks unused

**Review by martin-g — CHANGES_REQUESTED** [review]

File: lang/rust/avro/src/schema.rs [hunk 2]
```diff
@@ -2030,6 +2030,51 @@ mod tests {
         assert_eq!(schema_c, schema_c_expected);
     }
 
+    // AVRO-3584 : recursion in type definitions
+    #[test]
+    fn test_recursion_records() {
```
[the remark this answers is not in the document]
Comment (clesaec) [reply lang/rust/avro/src/schema.rs]: updated

File: lang/rust/avro/src/schema.rs (hunk 1 above)
Comment (clesaec) [reply lang/rust/avro/src/schema.rs]: removed

File: lang/rust/avro/src/schema.rs [hunk 3]
```diff
@@ -2030,6 +2030,44 @@ mod tests {
...
+        let schema_a = list
+            .first()
+            .unwrap()
+            .clone();
+
+        match schema_a {
+            Schema::Record { fields, .. } => {
+                let f1 = fields.get(0);
+
+                let refSchema = Schema::Ref {
```
Comment (martin-g) [inline lang/rust/avro/src/schema.rs:2061]: in the previous run `clippy` complained that `refSchema` should be `ref_schema`, but it didn't complain in this run ...
Strange.

Comment (clesaec) [reply lang/rust/avro/src/schema.rs]: ok, renamed to ref_schema.

**Review by martin-g — APPROVED** [review]

**Comment by martin-g** [comment]
Thanks for the new test, @clesaec !

But please do something about the `merge commits` for the next PR! :-)

**Pull request merged**